fedimint-server/src/net/api.rs

```diff
@@ -342,12 +342,21 @@ impl ConsensusApi {
     async fn get_federation_audit(&self) -> ApiResult<AuditSummary> {
         let mut dbtx = self.db.begin_transaction().await;
         let mut audit = Audit::default();
-        for (module_instance_id, _, module) in self.modules.iter_modules() {
+        let mut module_instance_id_to_kind: HashMap<ModuleInstanceId, String> = HashMap::new();
+        for (module_instance_id, kind, module) in self.modules.iter_modules() {
+            module_instance_id_to_kind.insert(module_instance_id, kind.as_str().to_string());
             module
-                .audit(&mut dbtx.with_module_prefix(module_instance_id), &mut audit)
+                .audit(
+                    &mut dbtx.with_module_prefix(module_instance_id),
+                    &mut audit,
+                    module_instance_id,
+                )
                 .await
         }
-        Ok(AuditSummary::from_audit(&audit))
+        Ok(AuditSummary::from_audit(
+            &audit,
+            &module_instance_id_to_kind,
+        ))
```

Comment:
It's slightly annoying that we have to mutate two variables here, but I see how the pre-existing code makes this the most concise option to build the instance->kind map. I guess by inlining `from_audit` we could remove the need for it, but then this fn might be doing too much.